File: Easy/1512_NumberOfGoodPairs/Python3/Solution.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        m = set()
        c = 0
        for n in nums:
            if n not in m:
                c += self.countAllPossiblePairs(nums.count(n))
                m.add(n)
        return c
    
    def countAllPossiblePairs(self, k: int) -> int:
        if k < 2: return 0
        if k == 2: return 1
        return k - 1 + self.countAllPossiblePairs(k - 1)

File: Easy/1512_NumberOfGoodPairs/C#/Solution.cs
// using System.Collections.Generic;

public class Solution {
    public int NumIdenticalPairs(int[] nums) {
        int numberOfGoodPairs = 0;

        // for(int i = 0; i < nums.Length - 1; i++) {
        //     for(int j = i + 1; j < nums.Length; j++) {
        //         if(nums[i] == nums[j]) {
        //             numberOfGoodPairs++;
        //         }
        //     }
        // }

        int[] counter = new int[101];

        foreach(int number in nums) {
            numberOfGoodPairs += counter[number];
            counter[number]++;
        }

        // for(int i = 0; i < nums.Length; i++) {
        //     int number = nums[i];
        //     numberOfGoodPairs += counter[number];
        //     counter[number] += 1;
        // }

        // Dictionary<int, int> counter = new();


        // for(int i = 0; i < nums.Length; i++) {
        //     int number = nums[i];
        //     if (counter.TryGetValue(number, out int count)) {
        //         numberOfGoodPairs += count;
        //         counter[number] += 1;
        //     } else {
        //         counter[number] = 1;
        //     }
        // }

        return numberOfGoodPairs;
    }
}